website/docs/plugin/testing/acceptance-tests/configuration.mdx

```diff
@@ -0,0 +1,277 @@
+
+* [TestStep.Config](/terraform/plugin/testing/acceptance-tests/configuration#teststep-config)
+* [TestStep.ConfigDirectory](/terraform/plugin/testing/acceptance-tests/configuration#teststep-configdirectory)
+* [TestStep.ConfigFile](/terraform/plugin/testing/acceptance-tests/configuration#teststep-configfile)
+
+Terraform configuration can be used in conjunction with Terraform variables defined via [TestStep.ConfigVariables](/terraform/plugin/testing/acceptance-tests/configuration#teststep-configvariables).
+
+## TestStep Config
+
+The `TestStep.Config` field accepts a string containing valid Terraform configuration.
+
+In the following example, the `Config` field specifies a resource which is used in combination with `ExternalProviders` to specify the version and source for the provider:
+
+```go
+func TestAccResourcePassword_UpgradeFromVersion3_2_0(t *testing.T) {
+    resource.Test(t, resource.TestCase{
+        Steps: []resource.TestStep{
+            {
+                ExternalProviders: map[string]resource.ExternalProvider{
+                    "random": {
+                        VersionConstraint: "3.2.0",
+                        Source:            "hashicorp/random",
+                    },
+                },
+                Config: `resource "random_password" "min" {
+                            length = 12
+                            override_special = "!#@"
+                            min_lower = 2
+                            min_upper = 3
+                            min_special = 1
+                            min_numeric = 4
+                        }`,
+                        
+```
+
+## TestStep ConfigDirectory
+
+The `TestStep.ConfigDirectory` field accepts a [TestStepConfigFunc](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/config#TestStepConfigFunc) which is a function that accepts a [TestStepConfigRequest](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/config#TestStepConfigRequest) and returns a string containing a path to a directory containing Terraform configuration files. The path can be a relative or absolute path.
+
+There are helper methods available for generating a [TestStepConfigFunc](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/config#TestStepConfigFunc) including:
+
+* [StaticDirectory(directory string)](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/config#StaticDirectory)
+* [TestNameDirectory()](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/config#TestNameDirectory)
+* [TestStepDirectory()](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/config#TestStepDirectory)
+
+~> **Note**: `ExternalProviders` cannot be specified when using ConfigDirectory.
```

Comment:
Is the intention for developers to use the `required_providers` instead? Might be worth mentioning this doc: https://developer.hashicorp.com/terraform/language/providers/requirements#requiring-providers

Reply:
Good point. Have added a sentence to make this clearer and used the link you supplied. Thanks